app/views/layouts/_progress_bar.html.erb

```diff
@@ -1,12 +1,14 @@
 <div class="progress-list-flex">
-  <% guide.exercises.each do |e| %>
+  <% assignments = guide.assignments_for(current_user) %>
```

Comment:
Those variables are clean but have some drawbacks - they can't be reused in other partials of the same request. I my next PRs I am addressing this